### ISSUE TITLE
Math::Calc::Parser out of memory

### ISSUE THREAD

**Issue by Grinnz** [Grinnz/Math-Calc-Parser#3]
```
Hi,

I'm testing the Math::Calc::Parser module (newest version from CPAN) and
unfortunately there seems to be an easy way to trigger a out of memory
exception - so the whole script crashes - with e.g.

1000000000000!

interestingly, if I put up a number even higher, like

10000000000000000000!

I get a proper error "Error in function "!": Range iterator outside 
integer range"
but in the former case, it seems the module still attempts to solve the 
task
and fails (on my 16GB RAM notebook).


-- 
Rico
```

**Comment by Grinnz** [comment]
The problem here is likely forming the list to pass to reduce. The higher number is disallowed by the range operator itself so it doesn't get that far.

**Comment by Grinnz** [comment]
This should be fixed in 1.005. The new method iterates rather than prepopulating a list, and so both example cases now result in `Inf`.

**Comment by rico666** [comment]
I can confirm the bug is gone with 1.005
